### PR TITLE
Remove compositions and update readonly route type

### PR DESCRIPTION
**Pull request by pleek91**
# Description
Step one in making params writable. Updating params does not update the route but the tyypes and implementation allow for updating individual params (while still keeping the rest of the route object readonly)

### CODE REVIEW

File: src/types/resolved.ts [hunk 1]
```diff
@@ -8,4 +9,6 @@ export type ResolvedRoute<TRoute extends Route = Route> = {
   key: TRoute['key'],
   query: ResolvedRouteQuery,
   params: ExtractRouteParamTypes<TRoute>,
-}
\ No newline at end of file
+}
+
+export type ReadonlyResolvedRoute<TRoute extends ResolvedRoute = ResolvedRoute> = DeepReadonly<Omit<TRoute, 'params'>> & Readonly<{ params: TRoute['params'] }>
```
Comment (pleek91) [inline src/types/resolved.ts:14]: I'm not sold on 
1. Introducing another route type
2. The name.

Comment (stackoverfloweth) [reply src/types/resolved.ts:14]: Should ResolvedRoute be read only by default?

**Review by stackoverfloweth — APPROVED** [review]

File: src/types/resolved.ts (hunk 1 above)
Comment (pleek91) [reply src/types/resolved.ts:14]: Possibly? Once we resolve a route do we ever need to update it internally?

Comment (stackoverfloweth) [reply src/types/resolved.ts:14]: I think that's fine, I would proceed with that assumption and we can pivot in the future if we need to

**Review by stackoverfloweth — APPROVED** [review]

**Pull request merged**